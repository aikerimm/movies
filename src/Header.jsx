import React from 'react';
import logo from './assets/logo.svg';

const App = () => (
  <header>
    <nav className="nav">
      <img src={logo} alt="react-logo" className="nav-logo" />
      <ul className="nav-items">
        <li>Pricing</li>
        <li>About</li>
        <li>Contact</li>
      </ul>
    </nav>
  </header>);

export default App;
