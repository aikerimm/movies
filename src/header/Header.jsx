import React, { useCallback, useState } from 'react';
import './header.css';
import MovieModal from '../modal/MovieModal.jsx';
import Button from '../util/Button';
import AddMovieSuccessModal from '../modal/AddMovieSuccessModal.jsx';
import { sendAddMovieRequest } from '../util/apiService';
import FindYourMovie from './FindYourMovie';

const Header = () => {
  const [openAddMovieModal, setOpenAddMovieModal] = useState(false);
  const [openSuccessModal, setOpenSuccessModal] = useState(false);

  const handleOpenAddMovieModal = useCallback(
    () => setOpenAddMovieModal(true),
    []
  );
  const handleCloseAddMovieModal = useCallback(
    () => setOpenAddMovieModal(false),
    []
  );
  const handleAddMovieFormSubmit = useCallback((values) => {
    return sendAddMovieRequest(values).then((response) => {
      if (response.status == 201) {
        setOpenSuccessModal(true);
      } else {
        alert('Error adding movie. See network tab.');
      }
    });
  }, []);
  const handleAddMovieSuccessModalClose = useCallback(() => {
    setOpenSuccessModal(false);
    setOpenAddMovieModal(false);
    window.location.reload();
  }, []);

  return (
    <div className='header'>
      <div className='headerTop'>
        <p className='headerTitle'>
          <b>netflix</b>roulette
        </p>
        <Button
          value='+ Add movie'
          type='add'
          onClick={handleOpenAddMovieModal}
        />
      </div>
      <FindYourMovie />
      {openAddMovieModal && (
        <MovieModal
          onClose={handleCloseAddMovieModal}
          modalTitle='Add movie'
          onSubmit={handleAddMovieFormSubmit}
        />
      )}
      {openSuccessModal && (
        <AddMovieSuccessModal onClose={handleAddMovieSuccessModalClose} />
      )}
    </div>
  );
};

export default Header;
