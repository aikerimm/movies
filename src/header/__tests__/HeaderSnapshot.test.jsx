/**
 * @jest-environment jsdom
 */

import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../Header';

it('Header renders correctly', () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
