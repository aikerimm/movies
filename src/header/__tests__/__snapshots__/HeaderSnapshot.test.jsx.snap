// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="header"
  id="searchHeader"
>
  <div
    className="headerTop"
  >
    <p
      className="headerTitle"
    >
      <b>
        netflix
      </b>
      roulette
    </p>
    <input
      className="addBtn roundedCorners"
      onClick={[Function]}
      type="button"
      value="+ Add movie"
    />
  </div>
  <div
    className="findYourMovie"
  >
    <p
      className="titleText fullWidth"
    >
      Find your movie
    </p>
    <div
      className="search"
    >
      <input
        className="searchInput roundedCorners"
        id="searchInput"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="What do you want to watch?"
        type="text"
      />
      <input
        className="submitBtn roundedCorners"
        onClick={[Function]}
        type="button"
        value="search"
      />
    </div>
  </div>
</div>
`;
